### PR TITLE
feat: support the `accept` configuration on actions

### PR DESCRIPTION
**Pull request by transhapHigsn**
### Contributor checklist

- [ ] My commit messages follow the [Conventional Commit Message Format](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#format-of-the-commit-message)
      For example: `fix: Multiply by appropriate coefficient`, or
      `feat(Calculator): Correctly preserve history`
      Any explanation or long form information in your commit message should be
      in a separate paragraph, separated by a blank line from the primary message
- [ ] Bug fixes include regression tests
- [x] Features include unit/acceptance tests

### CODE REVIEW

**Comment by transhapHigsn** [comment]
There are still relationship changes remaining. Also test cases for patch and relationships remaining.

File: lib/ash_json_api/json_schema/json_schema.ex [hunk 1]
```diff
@@ -648,30 +648,32 @@ defmodule AshJsonApi.JsonSchema do
...
+    |> Enum.filter(&(is_nil(accept) || &1.name in accept))
     |> Enum.filter(& &1.writable?)
     |> Enum.reject(& &1.allow_nil?)
     |> Enum.reject(& &1.default)
     |> Enum.reject(& &1.generated?)
     |> Enum.map(&to_string(&1.name))
   end
 
-  defp write_attributes(resource) do
+  defp write_attributes(resource, accept \\ []) do
```
Comment (zachdaniel) [inline lib/ash_json_api/json_schema/json_schema.ex:664]: Why does this need to be optional?

**Comment by zachdaniel** [comment]
@transhapHigsn looking good so far!

File: lib/ash_json_api/json_schema/json_schema.ex (hunk 1 above)
Comment (transhapHigsn) [reply lib/ash_json_api/json_schema/json_schema.ex:664]: @zachdaniel I will be trimming everything extra out with further changes, this is probably due to fact that I started out with post changes first. I will have to check with other references of write_attributes as well.

File: lib/ash_json_api/json_schema/json_schema.ex [hunk 2]
```diff
@@ -594,8 +600,14 @@ defmodule AshJsonApi.JsonSchema do
...
   end
 
-  defp route_in_schema(%{type: type, relationship: relationship}, _api, resource)
+  defp route_in_schema(
+         %{type: type, relationship: relationship, action: action, action_type: action_type},
+         _api,
+         resource
+       )
        when type in [:post_to_relationship, :patch_relationship, :delete_from_relationship] do
+    accept = Ash.Resource.action(resource, action, action_type).accept
```
Comment (zachdaniel) [inline lib/ash_json_api/json_schema/json_schema.ex:609]: We won't want to use `accept` here actually. This is for adding/removing related entities. We can leave this one the way it is for now.

Comment (transhapHigsn) [reply lib/ash_json_api/json_schema/json_schema.ex:609]: will revert it then.

File: test/acceptance/post_test.exs [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+      resource(Author)
+    end
+  end
+
+  import AshJsonApi.Test
+
+  @tag :attributes
+  describe "invalid_post" do
+    test "create without all attributes in accept list" do
+      # dummy response.
```
Comment (zachdaniel) [inline test/acceptance/post_test.exs:114]: do we still need this?

Comment (transhapHigsn) [reply test/acceptance/post_test.exs:114]: no, it was just for comparison. removing it.

**Pull request merged**